Sources/Core/Client.swift

```diff
@@ -664,27 +663,53 @@ public actor Client {
         RunLoop.main.add(timer, forMode: .common)
     }
 
-    private func doSyncLoop() async {
-        guard self.isActive else {
-            Logger.debug("[SL] c:\"\(self.key)\" exit sync loop")
-            return
+    @discardableResult
+    private func performSyncInternal(_ isRealtimeSync: Bool, _ attachment: Attachment? = nil, _ syncMode: SyncMode = .pushPull) async throws -> [Document] {
+        await self.syncSemaphore.wait()
```

Comment:
Is there any reason for using AsyncSemaphore instead of DispatchSemaphore?

Reply:
performSyncInternal() is an async function. But DispatchSemaphore suspends the current thread. So If we use dispatchSemaphore, We have to create a new thread(aka global DispatchQueue) for the wait() function.
It causes thread explosion and is not work well with async/await basis.

AsyncSemaphore is task basis implementation of semaphore. So wait() works as a yield it's turn to another task, not blocking the threads.